cpp/include/raft/matrix/shift.cuh

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/matrix/detail/shift.cuh>
+
+namespace raft::matrix {
+
+/**
+ * @brief col_shift: in-place shifts all columns by k columns to the right and fills the first k
+ * columns in with "val"
+ * @param[in] handle: raft handle
+ * @param[in out] in_out: input matrix of size (n_rows, n_cols)
+ * @param[in] val: value to fill in the first k columns (same for all rows)
+ * @param[in] k: shift size
+ */
+template <typename math_t, typename matrix_idx_t>
+void col_right_shift(raft::resources const& handle,
```

Comment:
This naming is a bit confusing. Could we please instead have a "shift" function with an enum for "Direction::Left", "Direction::Right" and maybe use another enum for ShiftType::Row/ShiftType::Column? That would clean this up and minimize the number of free functions the user has to worry about invoking.

Reply:
Thank you for the suggestion, totally agree.
We have `towards_end` and `towards_zero` as the shift direction because left/right doesn't work with row shifting.

Reply:
Maybe `towards_start` and `towards_end`? `towards_zero` seems confusing (since we don't explicitly state that we're using indexing to determine the direction).